atom/browser/native_window_views_win.cc

```diff
@@ -151,6 +154,25 @@ bool NativeWindowViews::PreHandleMSG(
       if (w_param) {
         NotifyWindowEndSession();
       }
+      return false;
+    }
+    case WM_PARENTNOTIFY: {
+      if (LOWORD(w_param) == WM_CREATE) {
+        // Because of reasons regarding legacy drivers and stuff, a window that
+        // matches the client area is created and used internally by Chromium.
+        // This window is subclassed in order to fix some issues when forwarding
+        // mouse messages; see comments in |SubclassProc|. If by any chance
+        // Chromium removes the legacy window in the future it may be fine to
+        // move the logic to this very switch statement.
+        HWND legacy_window = reinterpret_cast<HWND>(l_param);
+        SetWindowSubclass(
+            legacy_window, SubclassProc, 1, reinterpret_cast<DWORD_PTR>(this));
+        if (legacy_window_map_.size() == 0) {
+          mouse_hook_ = SetWindowsHookEx(WH_MOUSE_LL, MouseHookProc, NULL, 0);
```

Comment:
This would install mouse hook for every Electron apps, we should only do subclass and install the hook when `win.setForwardMouseMessages` is called.